src/plugins/vis_types/vega/public/components/vega_vis_component.tsx

```diff
@@ -55,14 +55,14 @@ export const VegaVisComponent = ({
     }
   }, [visData, renderComplete]);
 
+  /* eslint-disable-next-line react-hooks/exhaustive-deps */
   const updateChartSize = useCallback(
-    () =>
-      throttle(() => {
-        if (visController.current) {
-          visController.current.render(visData).then(renderComplete);
-        }
-      }, 300),
-    [renderComplete, visData]
+    throttle(() => {
```

Comment:
@DziyanaDzeraviankina what if we do the following:
1. `src/plugins/vis_types/vega/public/vega_visualization.ts ` add new public method: 
  ```
 resize() {
      this.vegaView.resize();
    }
```
2. update the corresponding types
3. `src/plugins/vis_types/vega/public/components/vega_vis_component.tsx `
```
const updateChartSize = useMemo(
    () =>
      throttle(() => {
        visController.current?.resize();
      }, 300),
    []
  );

  useEffect(() => {
    return () => {
      updateChartSize.cancel();
    };
  }, [updateChartSize]);
```

Reply:
discussed offline